core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedDocServicePlugin.java

```diff
@@ -397,85 +445,42 @@ private static FieldLocation location(AnnotatedValueResolver resolver) {
         }
 
-        return newStructInfo(type);
-    }
-
-    private static EnumInfo newEnumInfo(Class<?> enumClass) {
-        final String name = enumClass.getName();
-        final Description description = AnnotationUtil.findFirst(enumClass, Description.class);
-
-        final Field[] declaredFields = enumClass.getDeclaredFields();
-        final List<EnumValueInfo> values =
-                Stream.of(declaredFields)
-                      .filter(Field::isEnumConstant)
-                      .map(f -> {
-                          final Description valueDescription = AnnotationUtil.findFirst(f, Description.class);
-                          if (valueDescription != null) {
-                              return new EnumValueInfo(f.getName(), null,
-                                                       DescriptionInfo.of(valueDescription.value(),
-                                                                          valueDescription.markup()));
-                          }
-
-                          return new EnumValueInfo(f.getName(), null);
-                      })
-                      .collect(toImmutableList());
-
-        if (description != null) {
-            return new EnumInfo(name, values, DescriptionInfo.of(description.value(), description.markup()));
-        }
-
-        return new EnumInfo(name, values);
-    }
-
-    private static StructInfo newStructInfo(Class<?> structClass) {
-        final String name = structClass.getName();
-        final Description description = AnnotationUtil.findFirst(structClass, Description.class);
-
-        final Field[] declaredFields = structClass.getDeclaredFields();
-        final List<FieldInfo> fields =
-                Stream.of(declaredFields)
-                      .map(f -> {
-                          final Description fieldDescription = AnnotationUtil.findFirst(f, Description.class);
-                          if (fieldDescription != null) {
-                              return FieldInfo.of(
-                                      f.getName(),
-                                      toTypeSignature(f.getGenericType()),
-                                      DescriptionInfo.of(fieldDescription.value(), fieldDescription.markup())
-                              );
-                          }
-
-                          return FieldInfo.of(f.getName(), toTypeSignature(f.getGenericType()));
-                      })
-                      .collect(toImmutableList());
-
-        if (description != null) {
-            return new StructInfo(name, fields, DescriptionInfo.of(description.value(), description.markup()));
+        // If the type can be serialized to JSON, try extracting a `StructInfo` using Jackson.
+        // Don't care about the request object because the `StructInfo` for the request object is processed
+        // when building the `FieldInfo` of the `MethodInfo`.
+        namedTypeInfo = defaultResponseNamedTypeInfoProvider.newNamedTypeInfo(typeDescriptor);
```

Comment:
Question) Do we assume that all `typeDescriptor`s that end up here are for `response` types? (I think `typeDescriptor`s could also be aggregated from `@Param`)

From reading the code, it seems like we don't know where the `typeDescriptor` is for (return type vs. parameter type).

Reply:
That was my mistake. As you said, a class defined with `@Param` could be injected from which Jackson can not extract useful field information.
`ReflectiveNamedTypeInfoProvider` is added to support for non-json object types.
Sorry for the frequent changes.